tck/src/main/java/org/eclipse/microprofile/rest/client/tck/interfaces/InvalidComputeMethodSignature.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2018 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.
+ *
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.eclipse.microprofile.rest.client.tck.interfaces;
+
+import java.util.Date;
+import javax.ws.rs.GET;
+import javax.ws.rs.client.ClientRequestContext;
+import javax.ws.rs.core.Response;
+import org.eclipse.microprofile.rest.client.annotation.ClientHeaderParam;
+
+public interface InvalidComputeMethodSignature {
+
+    default String invalidMethod(String headerName, String someOtherArg) {
+        return "should not be invoked - too many String args";
+    }
+
+    default String invalidMethod(String headerName, Integer someOtherArg) {
+        return "should not be invoked - unexpected Integer arg";
+    }
+
+    default String invalidMethod(ClientRequestContext context, ClientRequestContext someOtherArg) {
```

Comment:
should it work with single `ClientRequestContext`?
I cannot find a test that would check it.

Reply:
I was originally thinking that we could pass in a `ClientRequestContext` in addition the header name, but ultimately decided against it since a lot of the info in CRC would be invalid (or at least, not yet determined this early in the flow).  

I should remove that method from the TCK interface.

Is there anything in particular that you thought might be useful from the CRC?  We could possibly pass in the `Method` object from the client interface - that might allow compute methods to read other annotations, etc to better determine what the header value should be...

Reply:
I didn't have anything specific in mind.